lib/recog/nizer.rb

```diff
@@ -20,6 +20,39 @@ class Nizer
   }
 
   @@db_manager = nil
+  @@db_sorted = false
+
+
+  #
+  # Load fingerprints from a specific file or directory
+  # This will not preserve any fingerprints that have already been loaded
+  # @param path [String] Path to file or directory of XML fingerprints
+  def self.load_db(path=nil)
+    if path
+      @@db_manager = Recog::DBManager.new(path)
+    else
+      @@db_manager = Recog::DBManager.new
+    end
+
+    # Sort the databases, no behavior or result change for those calling
+    # Nizer.match or Nizer.multi_match as they have a single DB
+    @@db_manager.databases.sort! { |a,b| a.priority <=> b.priority }
+    @@db_sorted = true
+  end
+
+  #
+  # Display the fingerprint databases in the order in which they will be used
+  # to match banners.  This is useful for fingerprint tuning and debugging.
+  def self.display_db_order()
+    unless @@db_manager
+      @@db_manager = Recog::DBManager.new
+      @@db_manager.databases.sort! { |a,b| a.priority <=> b.priority }
```

Comment:
My thoughts were that display_db_order should only called infrequently and during debugging so sorting an already sorted list shouldn't cause any meaningful performance impact. There are other cases were attempts to sort the database list would be made every time match_all_db or multi_match_all_db were called. In those cases, despite the shortness of the array I tried to avoid sorting them.